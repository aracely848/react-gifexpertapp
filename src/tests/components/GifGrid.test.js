import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Prueba en el <GifGrid />', ()=> {

    const category = 'One Punch';


    test('Debe de mostrarse correctamente', ()=> {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        
        const wrapper = shallow( <GifGrid category={category}/>);        
        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', ()=> {
        const gifs = [{
            id:'ABC',
            url: 'https://localhostcualquier/cosa.jpg',
            title: 'Cualquiercosa'
        },
        {
            id:'123',
            url: 'https://localhostcualquier/cosa.jpg',
            title: 'Cualquiercosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
       
    })
})