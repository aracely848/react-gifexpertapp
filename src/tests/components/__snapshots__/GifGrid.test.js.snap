// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquiercosa"
      url="https://localhostcualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquiercosa"
      url="https://localhostcualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba en el <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
